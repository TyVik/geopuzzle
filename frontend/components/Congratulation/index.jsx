'use strict';
import React from "react";
import {Modal} from 'react-bootstrap';
import localization from '../../localization';


class Congratulation extends React.Component {
    constructor(props) {
        super(props);
        this.state = {...window.__CONGRATULATION__, show: true};
    }

    share_fb = () => {
        FB.ui({
            app_id: 1273749826026102,
            method: 'feed',
            display: 'popup',
            link: this.props.url,
            caption: this.props.share
        }, function(response){});
    };

    share_google = () => {
        window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=600,width=600');
    };

    onClose = () => {
        this.setState({...this.state, show: false});
    };

    render() {
        let text = this.state.text + this.props.result + '.';
        return <Modal show={this.state.show} onHide={this.onClose}
               bsSize="large" aria-labelledby="contained-modal-title-lg">
            <Modal.Header closeButton>
                <Modal.Title id="contained-modal-title-lg">{localization.congratulations}</Modal.Title>
            </Modal.Header>
            <Modal.Body>{text}</Modal.Body>
            <Modal.Footer>
                <div className="pull-right">
                    <a className="btn btn-social-icon btn-vk" target="_blank"
                       href={"https://vk.com/share.php?url=" + this.props.url + "&title=" + text}><span className="fa fa-vk" /></a>
                    <a className="btn btn-social-icon btn-facebook"
                       href="#" onClick={this.share_fb}
                       target="_blank"><span className="fa fa-facebook" /></a>
                    <a className="btn btn-social-icon btn-twitter"
                       href={"https://twitter.com/intent/tweet?text=" + text + "&url=" + this.props.url + "&hashtags=geopuzzle"}
                       target="_blank"><span className="fa fa-twitter" /></a>
                    <a href={"https://plus.google.com/share?url=" + this.props.url } onClick={this.share_google}><img
src="https://www.gstatic.com/images/icons/gplus-32.png" alt="Share on Google+"/></a>
                </div>
            </Modal.Footer>
        </Modal>;
    }
}


export default Congratulation;
