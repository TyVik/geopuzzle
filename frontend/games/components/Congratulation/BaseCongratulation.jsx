'use strict';
import React from "react";
import {Button, Modal} from 'react-bootstrap';
import {FormattedMessage as Msg} from "react-intl";

import "./index.css";


class BaseCongratulation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {show: true, url: location.href};
  }

  shareFb = () => {
    FB.ui({
      app_id: 1273749826026102,
      method: 'feed',
      display: 'popup',
      link: this.state.url,
      caption: this.props.share
    }, function(response){});
  };

  onClose = () => {
    this.setState(state => ({...state, show: false}));
  };

  getText() {
    let time = new Date(this.props.options.score * 1000);
    let params = {'name': window.__GAME__.name, 'subjects': window.__GAME__.parts, 'time': time.toLocaleTimeString('ru-RU', {timeZone: 'UTC'})};
    return this.props.intl.formatMessage({id: this.props.text}, params);
  }

  header() {
    return <Msg id="congratulations"/>;
  }

  render() {
    let text = this.getText();
    return <Modal show={this.state.show} onHide={this.onClose} aria-labelledby="contained-modal-title-lg">
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-lg">{this.header()}</Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <div>{text}</div>
      </Modal.Body>
      <Modal.Footer>
        <iframe src="https://www.patreon.com/platform/iframe?widget=become-patron-button&amp;redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&amp;creatorID=11576139"
                className="patreon-widget"></iframe>
        <div className="pull-right">
          <a className="btn btn-social-icon btn-vk" target="_blank" rel="noopener noreferrer"
             href={"https://vk.com/share.php?url=" + this.state.url + "&title=" + text}>
            <i className="fab fa-vk" />
          </a>
          <a className="btn btn-social-icon btn-facebook"
             href="#" onClick={this.shareFb}
             target="_blank" rel="noopener noreferrer">
            <i className="fab fa-facebook" />
          </a>
          <a className="btn btn-social-icon btn-twitter"
             href={"https://twitter.com/intent/tweet?text=" + text + "&url=" + this.state.url + "&hashtags=geopuzzle"}
             target="_blank" rel="noopener noreferrer">
            <i className="fab fa-twitter" />
          </a>
        </div>
      </Modal.Footer>
    </Modal>;
  }
}


export default BaseCongratulation;
