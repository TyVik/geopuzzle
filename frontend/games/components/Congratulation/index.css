.modal-body .additional-text {
    margin: 5px;
    font-weight: bold;
    text-align: center;
}
