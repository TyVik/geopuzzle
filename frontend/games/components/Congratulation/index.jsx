'use strict';
import React from "react";
import {Modal} from 'react-bootstrap';
import {decode} from 'he';
import {FormattedMessage as Msg} from "react-intl";


class Congratulation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {...window.__CONGRATULATION__, show: true};
  }

  share_fb = () => {
    FB.ui({
      app_id: 1273749826026102,
      method: 'feed',
      display: 'popup',
      link: this.props.url,
      caption: this.props.share
    }, function(response){});
  };

  share_google = () => {
    window.open(this.href, '', 'menubar=no,toolbar=no,resizable=yes,scrollbars=yes,height=600,width=600');
  };

  onClose = () => {
    this.setState({...this.state, show: false});
  };

  render() {
    let text = decode(this.state.text) + this.props.result + '.';
    return <Modal show={this.state.show} onHide={this.onClose} aria-labelledby="contained-modal-title-lg">
      <Modal.Header closeButton>
        <Modal.Title id="contained-modal-title-lg"><Msg id="congratulations"/></Modal.Title>
      </Modal.Header>
      <Modal.Body>{text}</Modal.Body>
      <Modal.Footer>
        <div className="pull-right">
          <a className="btn btn-social-icon btn-vk" target="_blank"
             href={"https://vk.com/share.php?url=" + this.props.url + "&title=" + text}>
            <i className="fab fa-vk" />
          </a>
          <a className="btn btn-social-icon btn-facebook"
             href="#" onClick={this.share_fb}
             target="_blank">
            <i className="fab fa-facebook" />
          </a>
          <a className="btn btn-social-icon btn-twitter"
             href={"https://twitter.com/intent/tweet?text=" + text + "&url=" + this.props.url + "&hashtags=geopuzzle"}
             target="_blank">
            <i className="fab fa-twitter" />
          </a>
        </div>
      </Modal.Footer>
    </Modal>;
  }
}


export default Congratulation;
