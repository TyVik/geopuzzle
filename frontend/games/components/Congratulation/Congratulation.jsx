'use strict';
import {injectIntl} from "react-intl";
import BaseCongratulation from "./BaseCongratulation";


export default injectIntl(BaseCongratulation);
