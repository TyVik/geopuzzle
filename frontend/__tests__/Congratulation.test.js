import React from 'react';
import {shallow} from 'enzyme';
import Congratulation from '../games/components/Congratulation';


describe('shallow <Congratulation /> components', () => {
  let props = {result: 'result', url: 'url'};
  let wrapper;
  window.__CONGRATULATION__ = {text: 'text'};

  beforeEach(() => {
    wrapper = shallow(<Congratulation {...props}/>);
  });

  it('render', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('click close', () => {
    wrapper.instance().onClose();
    expect(wrapper.html()).toBe('');
  });
});
