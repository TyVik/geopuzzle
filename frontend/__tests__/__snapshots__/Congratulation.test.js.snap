// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow <Congratulation /> components render 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <injectIntl(BaseCongratulation)
    name="Belarus"
    options={
      Object {
        "score": 60,
      }
    }
    parts={Array []}
    text="congratulations.puzzle"
  >
    <BaseCongratulation
      intl={
        Object {
          "defaultFormats": Object {},
          "defaultLocale": "en",
          "defaultRichTextElements": undefined,
          "formatDate": [Function],
          "formatDateTimeRange": [Function],
          "formatDateToParts": [Function],
          "formatDisplayName": [Function],
          "formatList": [Function],
          "formatListToParts": [Function],
          "formatMessage": [Function],
          "formatNumber": [Function],
          "formatNumberToParts": [Function],
          "formatPlural": [Function],
          "formatRelativeTime": [Function],
          "formatTime": [Function],
          "formatTimeToParts": [Function],
          "formats": Object {},
          "formatters": Object {
            "getDateTimeFormat": [Function],
            "getDisplayNames": [Function],
            "getListFormat": [Function],
            "getMessageFormat": [Function],
            "getNumberFormat": [Function],
            "getPluralRules": [Function],
            "getRelativeTimeFormat": [Function],
          },
          "locale": "en",
          "messages": Object {},
          "onError": [Function],
          "textComponent": Symbol(react.fragment),
          "timeZone": undefined,
          "wrapRichTextChunksInFragment": undefined,
        }
      }
      name="Belarus"
      options={
        Object {
          "score": 60,
        }
      }
      parts={Array []}
      text="congratulations.puzzle"
    >
      <Modal
        animation={true}
        aria-labelledby="contained-modal-title-lg"
        autoFocus={true}
        backdrop={true}
        dialogAs={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "ModalDialog",
            "render": [Function],
          }
        }
        enforceFocus={true}
        keyboard={true}
        onHide={[Function]}
        restoreFocus={true}
        show={true}
      >
        <Modal
          autoFocus={true}
          backdrop={true}
          backdropTransition={[Function]}
          containerClassName="modal-open"
          enforceFocus={true}
          keyboard={true}
          manager={
            BootstrapModalManager {
              "containers": Array [
                <body
                  class="modal-open"
                  style="padding-right: 0px;"
                >
                  <div
                    class="fade modal-backdrop show"
                  />
                  <div
                    aria-labelledby="contained-modal-title-lg"
                    aria-modal="true"
                    class="fade modal show"
                    role="dialog"
                    style="display: block;"
                    tabindex="-1"
                  >
                    <div
                      class="modal-dialog"
                    >
                      <div
                        class="modal-content"
                      >
                        <div
                          class="modal-header"
                        >
                          <div
                            class="modal-title h4"
                            id="contained-modal-title-lg"
                          >
                            congratulations
                          </div>
                          <button
                            class="close"
                            type="button"
                          >
                            <span
                              aria-hidden="true"
                            >
                              ×
                            </span>
                            <span
                              class="sr-only"
                            >
                              Close
                            </span>
                          </button>
                        </div>
                        <div
                          class="modal-body"
                        >
                          <div>
                            congratulations.puzzle
                          </div>
                        </div>
                        <div
                          class="modal-footer"
                        >
                          <iframe
                            class="patreon-widget"
                            src="https://www.patreon.com/platform/iframe?widget=become-patron-button&redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&creatorID=11576139"
                          />
                          <div
                            class="pull-right"
                          >
                            <a
                              class="btn btn-social-icon btn-vk"
                              href="https://vk.com/share.php?url=about:blank&title=congratulations.puzzle"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-vk"
                              />
                            </a>
                            <a
                              class="btn btn-social-icon btn-facebook"
                              href="#"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-facebook"
                              />
                            </a>
                            <a
                              class="btn btn-social-icon btn-twitter"
                              href="https://twitter.com/intent/tweet?text=congratulations.puzzle&url=about:blank&hashtags=geopuzzle"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-twitter"
                              />
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                </body>,
              ],
              "data": Array [
                Object {
                  "classes": Array [
                    "modal-open",
                  ],
                  "modals": Array [
                    Object {
                      "add": [Function],
                      "backdrop": <div
                        class="fade modal-backdrop show"
                      />,
                      "dialog": <div
                        aria-labelledby="contained-modal-title-lg"
                        aria-modal="true"
                        class="fade modal show"
                        role="dialog"
                        style="display: block;"
                        tabindex="-1"
                      >
                        <div
                          class="modal-dialog"
                        >
                          <div
                            class="modal-content"
                          >
                            <div
                              class="modal-header"
                            >
                              <div
                                class="modal-title h4"
                                id="contained-modal-title-lg"
                              >
                                congratulations
                              </div>
                              <button
                                class="close"
                                type="button"
                              >
                                <span
                                  aria-hidden="true"
                                >
                                  ×
                                </span>
                                <span
                                  class="sr-only"
                                >
                                  Close
                                </span>
                              </button>
                            </div>
                            <div
                              class="modal-body"
                            >
                              <div>
                                congratulations.puzzle
                              </div>
                            </div>
                            <div
                              class="modal-footer"
                            >
                              <iframe
                                class="patreon-widget"
                                src="https://www.patreon.com/platform/iframe?widget=become-patron-button&redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&creatorID=11576139"
                              />
                              <div
                                class="pull-right"
                              >
                                <a
                                  class="btn btn-social-icon btn-vk"
                                  href="https://vk.com/share.php?url=about:blank&title=congratulations.puzzle"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    class="fab fa-vk"
                                  />
                                </a>
                                <a
                                  class="btn btn-social-icon btn-facebook"
                                  href="#"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    class="fab fa-facebook"
                                  />
                                </a>
                                <a
                                  class="btn btn-social-icon btn-twitter"
                                  href="https://twitter.com/intent/tweet?text=congratulations.puzzle&url=about:blank&hashtags=geopuzzle"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    class="fab fa-twitter"
                                  />
                                </a>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>,
                      "isTopModal": [Function],
                      "remove": [Function],
                      "setBackdropRef": [Function],
                      "setDialogRef": [Function],
                    },
                  ],
                  "overflowing": true,
                  "style": Object {
                    "overflow": "",
                    "paddingRight": "",
                  },
                },
              ],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "modals": Array [
                Object {
                  "add": [Function],
                  "backdrop": <div
                    class="fade modal-backdrop show"
                  />,
                  "dialog": <div
                    aria-labelledby="contained-modal-title-lg"
                    aria-modal="true"
                    class="fade modal show"
                    role="dialog"
                    style="display: block;"
                    tabindex="-1"
                  >
                    <div
                      class="modal-dialog"
                    >
                      <div
                        class="modal-content"
                      >
                        <div
                          class="modal-header"
                        >
                          <div
                            class="modal-title h4"
                            id="contained-modal-title-lg"
                          >
                            congratulations
                          </div>
                          <button
                            class="close"
                            type="button"
                          >
                            <span
                              aria-hidden="true"
                            >
                              ×
                            </span>
                            <span
                              class="sr-only"
                            >
                              Close
                            </span>
                          </button>
                        </div>
                        <div
                          class="modal-body"
                        >
                          <div>
                            congratulations.puzzle
                          </div>
                        </div>
                        <div
                          class="modal-footer"
                        >
                          <iframe
                            class="patreon-widget"
                            src="https://www.patreon.com/platform/iframe?widget=become-patron-button&redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&creatorID=11576139"
                          />
                          <div
                            class="pull-right"
                          >
                            <a
                              class="btn btn-social-icon btn-vk"
                              href="https://vk.com/share.php?url=about:blank&title=congratulations.puzzle"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-vk"
                              />
                            </a>
                            <a
                              class="btn btn-social-icon btn-facebook"
                              href="#"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-facebook"
                              />
                            </a>
                            <a
                              class="btn btn-social-icon btn-twitter"
                              href="https://twitter.com/intent/tweet?text=congratulations.puzzle&url=about:blank&hashtags=geopuzzle"
                              rel="noopener noreferrer"
                              target="_blank"
                            >
                              <i
                                class="fab fa-twitter"
                              />
                            </a>
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>,
                  "isTopModal": [Function],
                  "remove": [Function],
                  "setBackdropRef": [Function],
                  "setDialogRef": [Function],
                },
              ],
              "scrollbarSize": 0,
            }
          }
          onEnter={[Function]}
          onEntering={[Function]}
          onEscapeKeyDown={[Function]}
          onExit={[Function]}
          onExited={[Function]}
          onHide={[Function]}
          renderBackdrop={[Function]}
          renderDialog={[Function]}
          restoreFocus={true}
          show={true}
          transition={[Function]}
        >
          <Portal
            containerInfo={
              <body
                class="modal-open"
                style="padding-right: 0px;"
              >
                <div
                  class="fade modal-backdrop show"
                />
                <div
                  aria-labelledby="contained-modal-title-lg"
                  aria-modal="true"
                  class="fade modal show"
                  role="dialog"
                  style="display: block;"
                  tabindex="-1"
                >
                  <div
                    class="modal-dialog"
                  >
                    <div
                      class="modal-content"
                    >
                      <div
                        class="modal-header"
                      >
                        <div
                          class="modal-title h4"
                          id="contained-modal-title-lg"
                        >
                          congratulations
                        </div>
                        <button
                          class="close"
                          type="button"
                        >
                          <span
                            aria-hidden="true"
                          >
                            ×
                          </span>
                          <span
                            class="sr-only"
                          >
                            Close
                          </span>
                        </button>
                      </div>
                      <div
                        class="modal-body"
                      >
                        <div>
                          congratulations.puzzle
                        </div>
                      </div>
                      <div
                        class="modal-footer"
                      >
                        <iframe
                          class="patreon-widget"
                          src="https://www.patreon.com/platform/iframe?widget=become-patron-button&redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&creatorID=11576139"
                        />
                        <div
                          class="pull-right"
                        >
                          <a
                            class="btn btn-social-icon btn-vk"
                            href="https://vk.com/share.php?url=about:blank&title=congratulations.puzzle"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <i
                              class="fab fa-vk"
                            />
                          </a>
                          <a
                            class="btn btn-social-icon btn-facebook"
                            href="#"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <i
                              class="fab fa-facebook"
                            />
                          </a>
                          <a
                            class="btn btn-social-icon btn-twitter"
                            href="https://twitter.com/intent/tweet?text=congratulations.puzzle&url=about:blank&hashtags=geopuzzle"
                            rel="noopener noreferrer"
                            target="_blank"
                          >
                            <i
                              class="fab fa-twitter"
                            />
                          </a>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </body>
            }
          >
            <BackdropTransition
              appear={true}
              in={true}
            >
              <Fade
                appear={true}
                in={true}
                mountOnEnter={false}
                timeout={null}
                unmountOnExit={false}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={true}
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={null}
                  unmountOnExit={false}
                >
                  <div
                    className="fade modal-backdrop show"
                    onClick={[Function]}
                  />
                </Transition>
              </Fade>
            </BackdropTransition>
            <DialogTransition
              appear={true}
              in={true}
              onEnter={[Function]}
              onEntering={[Function]}
              onExit={[Function]}
              onExited={[Function]}
              unmountOnExit={true}
            >
              <Fade
                appear={true}
                in={true}
                mountOnEnter={false}
                onEnter={[Function]}
                onEntering={[Function]}
                onExit={[Function]}
                onExited={[Function]}
                timeout={null}
                unmountOnExit={true}
              >
                <Transition
                  addEndListener={[Function]}
                  appear={true}
                  enter={true}
                  exit={true}
                  in={true}
                  mountOnEnter={false}
                  onEnter={[Function]}
                  onEntered={[Function]}
                  onEntering={[Function]}
                  onExit={[Function]}
                  onExited={[Function]}
                  onExiting={[Function]}
                  timeout={null}
                  unmountOnExit={true}
                >
                  <div
                    aria-labelledby="contained-modal-title-lg"
                    aria-modal={true}
                    className="fade modal show"
                    onClick={[Function]}
                    onMouseUp={[Function]}
                    role="dialog"
                    style={
                      Object {
                        "paddingLeft": undefined,
                        "paddingRight": undefined,
                      }
                    }
                    tabIndex={-1}
                  >
                    <ModalDialog
                      onMouseDown={[Function]}
                    >
                      <div
                        className="modal-dialog"
                        onMouseDown={[Function]}
                      >
                        <div
                          className="modal-content"
                        >
                          <ModalHeader
                            closeButton={true}
                            closeLabel="Close"
                          >
                            <div
                              className="modal-header"
                            >
                              <ModalTitle
                                id="contained-modal-title-lg"
                              >
                                <ForwardRef
                                  className="modal-title"
                                  id="contained-modal-title-lg"
                                >
                                  <div
                                    className="modal-title h4"
                                    id="contained-modal-title-lg"
                                  >
                                    <FormattedMessage
                                      id="congratulations"
                                    >
                                      congratulations
                                    </FormattedMessage>
                                  </div>
                                </ForwardRef>
                              </ModalTitle>
                              <CloseButton
                                label="Close"
                                onClick={[Function]}
                              >
                                <button
                                  className="close"
                                  onClick={[Function]}
                                  type="button"
                                >
                                  <span
                                    aria-hidden="true"
                                  >
                                    ×
                                  </span>
                                  <span
                                    className="sr-only"
                                  >
                                    Close
                                  </span>
                                </button>
                              </CloseButton>
                            </div>
                          </ModalHeader>
                          <ModalBody>
                            <div
                              className="modal-body"
                            >
                              <div>
                                congratulations.puzzle
                              </div>
                            </div>
                          </ModalBody>
                          <ModalFooter>
                            <div
                              className="modal-footer"
                            >
                              <iframe
                                className="patreon-widget"
                                src="https://www.patreon.com/platform/iframe?widget=become-patron-button&redirectURI=https%3A%2F%2Fgeopuzzle.org%2Fpuzzle%2Fworkshop%2F&creatorID=11576139"
                              />
                              <div
                                className="pull-right"
                              >
                                <a
                                  className="btn btn-social-icon btn-vk"
                                  href="https://vk.com/share.php?url=about:blank&title=congratulations.puzzle"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    className="fab fa-vk"
                                  />
                                </a>
                                <a
                                  className="btn btn-social-icon btn-facebook"
                                  href="#"
                                  onClick={[Function]}
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    className="fab fa-facebook"
                                  />
                                </a>
                                <a
                                  className="btn btn-social-icon btn-twitter"
                                  href="https://twitter.com/intent/tweet?text=congratulations.puzzle&url=about:blank&hashtags=geopuzzle"
                                  rel="noopener noreferrer"
                                  target="_blank"
                                >
                                  <i
                                    className="fab fa-twitter"
                                  />
                                </a>
                              </div>
                            </div>
                          </ModalFooter>
                        </div>
                      </div>
                    </ModalDialog>
                  </div>
                </Transition>
              </Fade>
            </DialogTransition>
          </Portal>
        </Modal>
      </Modal>
    </BaseCongratulation>
  </injectIntl(BaseCongratulation)>
</IntlProvider>
`;
